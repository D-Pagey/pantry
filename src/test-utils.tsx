import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import React, { FC } from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { FirebaseContext } from './components/ProviderFirebase';

const AllTheProviders: FC = ({ children }) => (
    <FirebaseContext.Provider
        value={{
            categories: [],
            deleteFoodItem: () => () => null,
            expiringCount: 0,
            isAuthed: false,
            isCheckingAuth: false,
            fridge: [],
            signOut: () => null,
            updateFridge: () => null,
            user: {
                email: null,
                name: null
            }
        }}
    >
        <MemoryRouter>{children}</MemoryRouter>
    </FirebaseContext.Provider>
);

const customRender = (ui: any, options?: any) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
