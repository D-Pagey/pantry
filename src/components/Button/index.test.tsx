import React from 'react';
import userEvent from '@testing-library/user-event';
import { render } from '../../test-utils';
import { Button } from '.';

describe('Button component', () => {
    it('should render', () => {
        const { container } = render(<Button>Click me</Button>);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should call onClick once clicked', () => {
        const onClick = jest.fn();
        const name = 'Click me';
        const { getByText } = render(<Button onClick={onClick}>{name}</Button>);

        userEvent.click(getByText(name));

        expect(onClick).toHaveBeenCalled();
    });

    it('should have disabled attribute if passed disabled prop', () => {
        const { getByText } = render(<Button disabled>Click me</Button>);
        expect(getByText('Click me')).toBeDisabled();
    });

    it('should not call onClick function if disabled', () => {
        const onClick = jest.fn();
        const name = 'Click me';
        const { getByText } = render(
            <Button onClick={onClick} disabled>
                {name}
            </Button>
        );

        userEvent.click(getByText(name));

        expect(onClick).not.toHaveBeenCalled();
    });

    it('should render loading content when loading', () => {
        const loadingContent = 'I am loading';

        const { getByText } = render(
            <Button isLoading loadingContent={loadingContent}>
                Helloooooo
            </Button>
        );

        getByText(loadingContent);
    });
});
