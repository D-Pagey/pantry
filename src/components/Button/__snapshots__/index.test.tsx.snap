// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render 1`] = `
.c0 {
  background-color: #348841;
  border: 0;
  border-radius: 5px;
  color: white;
  cursor: pointer;
  padding: 1rem;
  min-width: 68px;
}

<button
  class="c0"
  type="button"
>
  Click me
</button>
`;
